cli/src/main.rs

```diff
@@ -276,6 +276,50 @@ mod tests {
 		assert!(execute(command).is_err());
 	}
 
+	#[test]
+	fn uint_encode_requires_decimal_inputs() {
+		let command = "ethabi encode params -v uint256 123abc --lenient".split(" ");
+		let result = execute(command);
+		assert!(result.is_err());
+		let err = result.unwrap_err();
+		assert_eq!(err.to_string(), "Ethabi error: Uint parse error: InvalidCharacter");
+	}
+
+	#[test]
+	fn uint_encode_big_numbers() {
+		let command = "ethabi encode params -v uint256 100000000000000000000000000000000022222222222222221111111111111 --lenient".split(" ");
+		let expected = "0000000000003e3aeb4ae1383562f4b82261d96a3f7a5f62ca19599c1ad6d1c7";
+		assert_eq!(execute(command).unwrap(), expected);
+	}
+
+	#[test]
+	fn int_encode_large_negative_numbers() {
+		// i256::min_value() is ok
+		let command = "ethabi encode params -v int256 -57896044618658097711785492504343953926634992332820282019728792003956564819967 --lenient".split(" ");
+		let expected = "8000000000000000000000000000000000000000000000000000000000000001";
+		assert_eq!(execute(command).unwrap(), expected);
+
+		// i256::min_value() -1 is too much
+		let command = "ethabi encode params -v int256 -57896044618658097711785492504343953926634992332820282019728792003956564819968 --lenient".split(" ");
+		assert_eq!(execute(command).unwrap_err().to_string(), "Ethabi error: int256 parse error: Underflow");
+	}
+
+	#[test]
+	fn int_encode_large_positive_numbers() {
+		// Overflow
+		let command = "ethabi encode params -v int256 100000000000000000000000000000000022222222222222221111111111111333333333344556 --lenient".split(" ");
+		assert_eq!(execute(command).unwrap_err().to_string(), "Ethabi error: int256 parse error: Overflow");
+
+		// Half of i256::max_value() is ok
```

Comment:
```suggestion
		// i256::max_value() is ok
```
the tests data should be adjusted by -1